core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1199,13 +1199,13 @@ public int size() {
      * @return Sorted list of {@link FingerprintFacet}s 
      */
     public @Nonnull Collection<FingerprintFacet> getSortedFacets() {
-        List<FingerprintFacet> r = new ArrayList<FingerprintFacet>(getFacets());
-        Collections.sort(r,new Comparator<FingerprintFacet>() {
+        List<FingerprintFacet> r = new ArrayList<>(getFacets());
+        r.sort(new Comparator<FingerprintFacet>() {
```

Comment:
Seems like you can likely use `Comparator.comparingLong(FingerprintFacet::getTimestamp)` instead of the manually written one.

Reply:
It looks like that happens [often in jenkins](https://github.com/jenkinsci/jenkins/search?q=%22new+Comparator%22&unscoped_q=%22new+Comparator%22), I might go back and add a commit later which does that cleanup: https://github.com/jenkinsci/jenkins/blob/22aa2e6e766074d11249893e3f35e0b99e20d3d0/test/src/test/java/jenkins/model/ExtensionTypeHierarchyTest.java#L55
https://github.com/jenkinsci/jenkins/blob/0795e89b308ec7fcbda097858d58763d8531be8c/core/src/main/java/hudson/views/MyViewsTabBar.java#L89
https://github.com/jenkinsci/jenkins/blob/d9aa1283e692f7fc1e19b8e360a971871ce6b31e/test/src/test/java/hudson/model/UsageStatisticsTest.java#L158

https://stackoverflow.com/questions/32995559/reverse-a-comparator-in-java-8 suggests `Comparator.reverseOrder()` might be able to handle:
https://github.com/jenkinsci/jenkins/blob/22aa2e6e766074d11249893e3f35e0b99e20d3d0/core/src/main/java/hudson/model/RunMap.java#L159

Reply:
Figures because this feature was added in Java 8, and you're the main person making Java 8 related updated to Jenkins at the moment. ;)